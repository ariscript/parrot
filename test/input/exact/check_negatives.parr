def length(l):
    if l == nil: 0
    else:
        let (_, rest) = l
        in 1 + length(rest)

check:
    length(nil) is 0,
    length((1, nil)) is== 1,
    length((1, (2, nil))) isn't 1.

0
