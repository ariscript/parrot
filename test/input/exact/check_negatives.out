check block: 3/3 tests passed
0
